Observer/Adminhtml/ActionPredispatch.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2020 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Observer\Adminhtml;
+
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+
+/**
+ * Observer for controller_action_predispatch
+ * @see \Magento\Framework\App\Action\Action::dispatch
+ */
+class ActionPredispatch implements ObserverInterface
+{
+    /**
+     * @var \Bolt\Boltpay\Model\EventsForThirdPartyModules
+     */
+    private $eventsForThirdPartyModules;
+
+    /**
+     * ActionPredispatch constructor.
+     * @param \Bolt\Boltpay\Model\EventsForThirdPartyModules $eventsForThirdPartyModules
+     */
+    public function __construct(\Bolt\Boltpay\Model\EventsForThirdPartyModules $eventsForThirdPartyModules)
+    {
+        $this->eventsForThirdPartyModules = $eventsForThirdPartyModules;
+    }
+
+    /**
+     * Redirects the predispatch event to the {@see \Bolt\Boltpay\Model\EventsForThirdPartyModules}
+     *
+     * @param Observer $observer
+     * @return void
+     */
+    public function execute(Observer $observer)
+    {
+        /** @var \Magento\Framework\App\RequestInterface $request */
+        $request = $observer->getData('request');
+        $this->eventsForThirdPartyModules->dispatchEvent(
```

Comment:
@danac-gs Looks like you added few observers but only one listener. Did I miss something?

Reply:
@vitaliyreznikov 
The additional event dispatches along with the one in use are there for future use.

The intention was to port the [core Magento 2 pre-dispatch functionality](https://github.com/magento/magento2/blob/5a9fcab6a35b6b39e10f47ca1ea22dfca4b794a4/lib/internal/Magento/Framework/App/Action/Plugin/EventDispatchPlugin.php#L108-L116) to our Events For Third-Party Modules. You can see that 3 events are dispatched there too, with varying degrees of specificity. 

Since we don't have area support in our implementation there is also a `adminhtml` prefix to each of them.
The two not used ones can, of course, be removed.